### ISSUE TITLE
Add support for react-router

### ISSUE THREAD

**Issue by vladfr** [okgrow/analytics#132]
Hello!

This library is great! Any +1s for built-in support for react-router?
Is there a way to make this work with react-router?

**Comment by cfnelson** [comment]
@vladfr Currently this package does not support react-router. You should still be able to use this package but you will not have any of the automatic page events being logged and will have to add them into the react-router yourself. 

We will be adding react-router support soon. We are looking at refactoring this package to be compatible with any router and not depend in any router packages.

Hope this helps 😄

**Comment by torayeff** [comment]
> You should still be able to use this package but you will not have any of the automatic page events being logged and will have to add them into the react-router yourself.

@cfnelson  Can you provide simple example for react-router?

**Comment by arrygoo** [comment]
@torayeff you could take a look at these for inspiration: https://github.com/okgrow/analytics/blob/master/client/flow-router.js#L7-L39
https://github.com/okgrow/analytics/blob/8e9421e723c2ac177d0c239af725553b0a8efe9d/client/helpers.js#L11-L18

Basically you would need to listen to route change and trigger a [`analytics.page` ](https://segment.com/docs/sources/website/analytics.js/#page) when the page is ready.

**Comment by pauldowman** [comment]
Really we need to refactor it to be router-independent. We definitely want to do that.  I think we might be able to simply listen for the `popstate` event and get rid of any router-specific code.

**Comment by Siyfion** [comment]
This would be a fantastic addition to the package, especially as more and more people are switching to React. I would assume that at least a portion of those users are also switching to React Router (such as myself).

**Comment by acomito** [comment]
I'm thinking of just popping it into an onEnter hook for now. Has anybody tried this? Do you have an example?

found this too: https://www.npmjs.com/package/react-mixpanel

**Comment by arrygoo** [comment]
@acomito I don't think we have tried it with react router yet, but I personally think you're on the right path with onEnter.

**Comment by pauldowman** [comment]
I think it could actually all be refactored to be router-independent, e.g. using the `popState` event.

But, I noticed it sounds like React Router might be implementing this soon. On the v4 branch they have this:

> We have some code close to being published that will manage the scroll positions of window and individual elements.

https://github.com/ReactTraining/react-router/tree/v4#what-about-scrolling

**Comment by pauldowman** [comment]
Actually, ignore that comment, it's on the wrong repo. :-)

**Comment by acomito** [comment]
So getting it (roughly) going was pretty easy:

setup settings per usual but with page views turned off:

```
"public": {
    "analyticsSettings": {
      "Mixpanel": {
        "token": YOUR_TOKEN_HERE,
        "people": true,
        "autorun": false
      }
    }
  }
```

Go into your routes file and add a function:

```
import { analytics } from 'meteor/okgrow:analytics'

const logPageView = (nextState, replace) => {
  console.log(nextState.location.pathname); //logs the pathname like '/refund'
  analytics.page(nextState.location.pathname);
}
```

then you can add it on a per route basis in onEnter:

`<Route name="refund" path="/refund" component={ RefundPolicy } onEnter={ logPageView } />`

Or even at the top-level route in the onChange and it will log for all child routes:

```
<Route path="/" component={ App } onChange={ logPageView }>
        <IndexRoute name="index" component={ Index } onChange={ () => window.scrollTo(0, 0) }  />
        <Route name="login" path="/login" component={ Login } />
</Route>
```

onChange has a slightly different signature to onEnter though, so change your function to:

```

const logPageView = (prevState, nextState, replace, ) => {
  console.log(nextState.location.pathname);
  analytics.page(nextState.location.pathname);
}

```

**Comment by ghost** [comment]
We've just released an update to this package (v2.1.0) which moves to a "router agnostic" approach (using the browser's history API). You might want to give this a try.
